### PR TITLE
Release/v1.24 - backport of https://github.com/envoyproxy/envoy/pull/24302

### LINKED ISSUE #24632: Please merge https://github.com/envoyproxy/envoy/pull/24302 into release 1.24
https://github.com/envoyproxy/envoy/pull/24302 was merged into main but can it be backported into release 1.24 so Istio can pick up the change sooner than later?

Thank you.

### PR DESCRIPTION
**Pull request by ceastman-r7**
Commit Message: attributes: implement xds info (#24302)
Additional Description: backport of PR https://github.com/envoyproxy/envoy/pull/24302

PR 24302:
Commit Message: Support look up of xDS info from expressions.
Additional Description: Promotes Wasm-specific attributes to first-class (usable in access log, for example), and reduces some code duplication.
Risk Level: low (new attributes)
Testing: unit
Docs Changes: yes
Release Notes: yes
Related: https://github.com/envoyproxy/envoy/pull/24023

Fix #24632

### CODE REVIEW

**Comment by phlax** [comment]
thanks for raising this - could you fix DCO please

**Comment by phlax** [comment]
cc @kyessenov

File: changelogs/current.yaml [hunk 1]
```diff
@@ -34,5 +34,116 @@ removed_config_or_runtime:
 # *Normally occurs at the end of the* :ref:`deprecation period <deprecated>`
 
 new_features:
+- area: tls
```
Comment (kyessenov) [inline changelogs/current.yaml:37]: This delta looks like a bad merge.

**Review by kyessenov — CHANGES_REQUESTED** [review]
LGTM apart from the bad changelog merge.
/wait-any

File: changelogs/current.yaml (hunk 1 above)
Comment (ceastman-r7) [reply changelogs/current.yaml:37]: How can I fix this? I am just trying to get https://github.com/envoyproxy/envoy/pull/24302 merged into the v1.24 release so Istio can use it.

Comment (kyessenov) [reply changelogs/current.yaml:37]: delete the text blob below, and only leave this section:
```
- area: attributes
  change: |
    added :ref:`attributes <arch_overview_attributes>` for looking up xDS configuration information.
```

**Review by kyessenov — APPROVED** [review]

**Comment by phlax** [comment]
requires #24988

**Comment by ceastman-r7** [comment]
I added the code change from https://github.com/envoyproxy/envoy/pull/24988

**Comment by phlax** [comment]
@ceastman-r7 its better not to force push as it makes it harder to review just the changes

**Comment by phlax** [comment]
> I was just following the recommended fix per the DCO document.

i see - no problem - it just meant that i couldnt just land it - it needs to be reviewed again

re DCO there are hooks described here https://github.com/envoyproxy/envoy/blob/main/CONTRIBUTING.md#submitting-a-pr that will auto add DCO

you might also want to do `echo NO_VERIFY=1 > .env` if you add the hooks but only want the DCO part

**Comment by phlax** [comment]
seems like it will need a branch merge again also

**Review by kyessenov — APPROVED** [review]

**Comment by phlax** [comment]
thanks for this @ceastman-r7

**Comment by ceastman-r7** [comment]
@phlax @kyessenov thank you for your help and patience.

**Pull request merged**